File: veins_test/zig/echo_args.zig
#!/usr/bin/env -S fragletc --vein=zig
pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();
    var args = try std.process.argsWithAllocator(arena.allocator());
    defer args.deinit();
    _ = args.next();
    var list = std.ArrayList(u8).init(arena.allocator());
    while (args.next()) |arg| {
        if (list.items.len > 0) try list.appendSlice(" ");
        try list.appendSlice(arg);
    }
    const stdout = std.io.getStdOut().writer();
    try stdout.print("Args: {s}\n", .{list.items});
}

File: veins_test/c/echo_args.c
#!/usr/bin/env -S fragletc --vein=c
#include <stdio.h>
int main(int argc, char *argv[]) {
    if (argc > 1) printf("First: %s\n", argv[1]);
    if (argc > 2) printf("Second: %s\n", argv[2]);
    return 0;
}
